x-pack/plugins/siem/public/components/page/hosts/first_last_seen_host/index.test.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { cloneDeep } from 'lodash/fp';
+import * as React from 'react';
+import { MockedProvider } from 'react-apollo/test-utils';
+import { render } from 'react-testing-library';
+
+import { mockFirstLastSeenHostQuery } from '../../../../containers/hosts/first_last_seen/mock';
+import { wait } from '../../../../lib/helpers';
+import { TestProviders } from '../../../../mock';
+
+import { FirstLastSeenHost } from '.';
+
+describe('FirstLastSeen Component', async () => {
+  // this is just a little hack to silence a warning that we'll get until react
+  // fixes this: https://github.com/facebook/react/pull/14853
+  // For us that mean we need to upgrade to 16.9.0
+  // and we will be able to do that when we are in master
+  // tslint:disable-next-line:no-console
+  const originalError = console.error;
+  beforeAll(() => {
+    // tslint:disable-next-line:no-console
+    console.error = (...args: string[]) => {
+      if (/Warning.*not wrapped in act/.test(args[0])) {
+        return;
+      }
+      originalError.call(console, ...args);
+    };
+  });
+
+  afterAll(() => {
+    // tslint:disable-next-line:no-console
+    console.error = originalError;
+  });
+
+  test('Loading', async () => {
+    const { container } = render(
+      <TestProviders>
+        <MockedProvider mocks={mockFirstLastSeenHostQuery} addTypename={false}>
+          <FirstLastSeenHost hostname="kibana-siem" type="first-seen" />
+        </MockedProvider>
+      </TestProviders>
+    );
+    expect(container.innerHTML).toBe(
+      '<div class="euiLoadingSpinner euiLoadingSpinner--medium"></div>'
+    );
+  });
+
```

Comment:
I'd add a test for `firstSeen` and `lastSeen` with each having an invalid date time to touch that part of your code so if you need to refactor it, you will have that warm fuzzy feeling you are ok.

Here are two that work for me. Sorry for the `any` cast but the type is only that is `object` at the moment.

```suggestion
  test('First Seen With a bad date time string', async () => {
    const badDateTime = cloneDeep(mockFirstLastSeenHostQuery);
    // tslint:disable-next-line:no-any
    (badDateTime[0] as any).result.data.source.HostFirstLastSeen.firstSeen = 'something-invalid';
    const { container } = render(
      <TestProviders>
        <MockedProvider mocks={badDateTime} addTypename={false}>
          <FirstLastSeenHost hostname="kibana-siem" type="first-seen" />
        </MockedProvider>
      </TestProviders>
    );

    await wait();

    expect(container.innerHTML).toBe('something-invalid');
  });

  test('Last Seen With a bad date time string', async () => {
    const badDateTime = cloneDeep(mockFirstLastSeenHostQuery);
    // tslint:disable-next-line:no-any
    (badDateTime[0] as any).result.data.source.HostFirstLastSeen.lastSeen = 'something-invalid';
    const { container } = render(
      <TestProviders>
        <MockedProvider mocks={badDateTime} addTypename={false}>
          <FirstLastSeenHost hostname="kibana-siem" type="last-seen" />
        </MockedProvider>
      </TestProviders>
    );

    await wait();

    expect(container.innerHTML).toBe('something-invalid');
  });
```

Reply:
I will add yours but this test https://github.com/elastic/kibana/pull/34831/files/5baedb2693a1e1eb92b21d19ef0222c865bb40c7#diff-59c2a42fc55e28612d1f7dde3c1ea5beR83 was doing the invalid date too